extension/js/common/composer/composer-send-btn.ts

```diff
@@ -20,441 +20,433 @@ declare const openpgp: typeof OpenPGP;
+    const currentActive = sendingOptions.filter('.active');
+    if (e.key === 'Escape') {
+      e.stopPropagation();
+      this.toggleSendOptions(e);
+    } else if (e.key === 'ArrowUp') {
+      e.preventDefault();
+      let prev = currentActive.prev();
+      if (!prev.length) {
+        prev = sendingOptions.last();
+      }
+      currentActive.removeClass('active');
+      prev.addClass('active');
+    } else if (e.key === 'ArrowDown') {
+      e.preventDefault();
+      let next = currentActive.next();
+      if (!next.length) {
+        next = sendingOptions.first();
+      }
+      currentActive.removeClass('active');
+      next.addClass('active');
+    } else if (e.key === 'Enter') {
+      e.stopPropagation();
+      e.preventDefault();
+      currentActive.click();
     }
+  }
 
-    private toggleSendOptions = (event: JQuery.Event<HTMLElement, null>) => {
-        event.stopPropagation();
-        const sendingContainer = $('.sending-container');
-        sendingContainer.toggleClass('popover-opened');
-        if (sendingContainer.hasClass('popover-opened')) {
-            $('body').click(Ui.event.handle((elem, event) => {
-                if (!this.composer.S.cached('sending_options_container')[0].contains(event.relatedTarget)) {
-                    sendingContainer.removeClass('popover-opened');
-                    $('body').off('click');
-                    this.composer.S.cached('toggle_send_options').off('keydown');
-                }
-            }));
-            this.composer.S.cached('toggle_send_options').on('keydown', Ui.event.handle(async (target, e) => this.sendingOptionsKeydownHandler(e)));
-            const sendingOptions = this.composer.S.cached('sending_options_container').find('.sending-option');
-            sendingOptions.hover(function () {
-                sendingOptions.removeClass('active');
-                $(this).addClass('active');
-            });
-        } else {
-            $('body').off('click');
-            this.composer.S.cached('toggle_send_options').off('keydown');
+  private extractProcessSendMsg = async () => {
+    this.composer.S.cached('toggle_send_options').hide();
+    try {
+      this.throwIfFormNotReady();
+      this.composer.S.now('send_btn_text').text('Loading');
+      Xss.sanitizeRender(this.composer.S.now('send_btn_i'), Ui.spinner('white'));
+      this.composer.S.cached('send_btn_note').text('');
+      const newMsgData = this.collectNewMsgData();
+      await this.throwIfFormValsInvalid(newMsgData);
+      const recipientsEmails = Array.prototype.concat.apply([], Object.values(newMsgData.recipients).filter(arr => !!arr)) as string[];
+      const senderKi = await this.app.storageGetKey(this.urlParams.acctEmail, this.composer.getSender());
+      const { armoredPubkeys, emailsWithoutPubkeys } = await this.app.collectAllAvailablePublicKeys(this.composer.getSender(), senderKi, recipientsEmails);
```

Comment:
This is actually not needed this high up, I'll move it down only for cases where there is something to encrypt